From 56b164c9585f78514128af9134b738a7045b9f39 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Mon, 21 Aug 2023 17:52:38 +0200
Subject: [PATCH 328/954] refactor(st-clock): code size optimization

Re-ordering structures to avoid gaps and minimize data.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git aaa4f5e861bfe99c6f684e4a4bfc73fce671d24a refactor(st-clock): code size optimization
890c6645d2a0b9febad381bdd8813d6aabc82c8c refactor(st-clock): code size optimization]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ic203c2f4ab208de74014c44100771505dc3daee0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/323782
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/clk/stm32mp2_clk.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/st/clk/stm32mp2_clk.c b/drivers/st/clk/stm32mp2_clk.c
index f950bd8e75..7121ca914d 100644
--- a/drivers/st/clk/stm32mp2_clk.c
+++ b/drivers/st/clk/stm32mp2_clk.c
@@ -30,19 +30,19 @@ uint32_t saved_pll_freq2_reg;
 
 struct stm32_osci_dt_cfg {
 	unsigned long freq;
+	uint32_t drive;
 	bool bypass;
 	bool digbyp;
 	bool css;
-	uint32_t drive;
 };
 
 struct stm32_pll_dt_cfg {
-	bool enabled;
+	uint32_t src;
+	uint32_t frac;
 	uint32_t cfg[PLLCFG_NB];
 	uint32_t csg[PLLCSG_NB];
-	uint32_t frac;
 	bool csg_enabled;
-	uint32_t src;
+	bool enabled;
 };
 
 struct stm32_clk_platdata {
@@ -507,7 +507,7 @@ static unsigned long clk_get_pll_fvco(struct stm32_clk_priv *priv,
 }
 
 struct stm32_pll_cfg {
-	int pll_id;
+	uint16_t pll_id;
 };
 
 static bool _clk_stm32_pll_is_enabled(struct stm32_clk_priv *priv, const struct stm32_clk_pll *pll)
@@ -712,7 +712,7 @@ static const struct stm32_clk_ops clk_stm32_pll1_ops = {
 }
 
 struct stm32_clk_flexgen_cfg {
-	int id;
+	uint8_t id;
 };
 
 static unsigned long clk_flexgen_recalc(struct stm32_clk_priv *priv, int idx,
-- 
2.53.0

